topeft/modules/datacard_tools.py

```diff
@@ -787,7 +787,14 @@ def analyze(self,km_dist,ch,selected_wcs, crop_negative_bins):
         outf_root_name = self.FNAME_TEMPLATE.format(cat=ch,kmvar=km_dist,ext="root")
 
         h = self.hists[km_dist]
+        try:
+            h_sumw2 = self.hists[km_dist+"_sumw2"]
+        except KeyError:
+            msg = "No sumw2 histogram found! Setting errors to 0"
+            print(msg)
+            h_sumw2 = None
         ch_hist = h.integrate("channel",[ch])
+        ch_sumw2 = h_sumw2.integrate("channel",[ch]) if h_sumw2 is not None else h_sumw2
```

Comment:
For this I wonder if we want to swap the order, i.e. have it instead be:
```
ch_sumw2 = h_sumw2 if h_sumw2 is None else h_sumw2.integrate("channel",[ch])
```
I'm sort of ambivalent on this, so I'm fine with keeping it as is. I just find it easier to mentally parse conditionals when there's as few negations as possible.

Reply:
I also like my suggested swap b/c (to me at least) the line reads more naturally. You make it clear that we leave `h_sumw2` "as is" if it's missing and if it is present do the usual integration over the "channel" axis that's happening in this section of the code.

Reply:
Done